ddsrouter_core/test/unittest/dynamic/discovery_database/DiscoveryDatabaseTest.cpp

```diff
@@ -28,6 +28,42 @@ using namespace eprosima::ddsrouter;
 using namespace eprosima::ddsrouter::core;
 using namespace eprosima::ddsrouter::core::types;
 
+namespace eprosima {
+namespace ddsrouter {
+namespace test {
+
+/**
+ * This class is used to expose protected methods of the parent class
+ * so they can be tested.
+ */
+class DiscoveryDatabase : public eprosima::ddsrouter::core::DiscoveryDatabase
+{
+public:
+
+    bool add_endpoint_protected(
+            const Endpoint& new_endpoint)
+    {
+        return add_endpoint_(new_endpoint);
+    }
+
+    bool update_endpoint_protected(
+            const Endpoint& new_endpoint)
+    {
+        return update_endpoint_(new_endpoint);
+    }
+
+    utils::ReturnCode erase_endpoint_protected(
+            const Guid& guid_of_endpoint_to_erase)
+    {
+        return erase_endpoint_(guid_of_endpoint_to_erase);
+    }
+
+};
+
+} /* namespace test */
+} /* namespace ddsrouter */
+} /* namespace eprosima */
+
```

Comment:
It would have been fine to actually check the new functionality :laughing:

Reply:
Just modified the communication tests to only use allowlist (empty builtin-topics), so now the dynamic topic discovery feature is tested in all of them.